### PR TITLE
[docs-only] Service Registry ADR

### PR DESCRIPTION
**Pull request by refs**

### CODE REVIEW

**Review by micbar — COMMENTED** [review]
Looks good in general, need to  deep-read

File: docs/ocis/adr/0006-service-discovery.md [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+
+* Status: proposed
+* Deciders: @refs, @butonic, @micbar, @dragotin, @pmaier1
+* Date: 2021-04-19
+
+Technical Story: [Introduce Named Services.](https://github.com/cs3org/reva/pull/1509)
+
+## Context and Problem Statement
+
+Reva relies heavily on config files. Implications of this approach are having to know a-priori where a service is running (host + port). We want to move away from hardcoded values and rely instead on named services for service discovery. Furthermore, we would like both platforms (Reva + oCIS) to have the same source of truth at any given time, not having one to notify the other whenever a service status changes.
```
Comment (micbar) [inline docs/ocis/adr/0006-service-discovery.md:19]: Implications of this approach ... -> only one impltcation, could be singular

File: docs/ocis/adr/0006-service-discovery.md [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+* Always up-to-date knowledge of the running services on a given deployment (a service registry doesn't have to necessarily be running on the same machine / network)
+
+## Considered Options
+
+* Hardcoded tuples of hostname + port
+* Dynamic service registration
+
+## Decision Outcome
+
+Chosen option: "Dynamic service registration". There were some drawbacks regarding this due to introducing go-micro to Reva was from start a problem, given how much of a bloat it would mean given the little usage of it. Instead we decided to define our very own [Registry interface](https://github.com/refs/reva/blob/58d013a7509d1941834e1bc814e9a9fa8bff00b1/pkg/registry/registry.go#L22-L35) on Reva and extended the runtime arguments to [allow for injecting a registry](https://github.com/refs/reva/blob/58d013a7509d1941834e1bc814e9a9fa8bff00b1/cmd/revad/runtime/option.go#L53-L58).
```
Comment (micbar) [inline docs/ocis/adr/0006-service-discovery.md:34]: Could split in two sentences

1) There were some drawbacks...
2) Introducing go-micro to Reva...

File: docs/ocis/adr/0006-service-discovery.md [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+
+### Positive Consequences
+
+* Having dynamic service registration delegates the entire lifecycle of finding a process to the service registry.
+* Removing a-priori knowledge of hostname + port for services.
+* Marrying go-micro's registry and a newly defined registry abstraction on Reva.
+* We will embrace go-micro interfaces by defining a third, merger interface in order to marry go-micro registry and rega revistry.
```
Comment (micbar) [inline docs/ocis/adr/0006-service-discovery.md:41]: unneccessary comma

File: docs/ocis/adr/0006-service-discovery.md [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+* The ability to fetch a service node relying only on its name (i.e: com.owncloud.proxy) and not on a tuple hostname + port that we rely on being preconfigured during runtime.
+* Conceptually speaking, a better framework to tie all the services together. Referring to services by names is less overall confusing than having to add a service name + where it is running. A registry is agnostic to "where is it running" because it, by definition, keeps track of this specific question, so when speaking about design or functionality, it will ease communication.
+
+## Pros and Cons of the Options
+
+### Hardcoded tuples of hostname + port
+
+* Good, because
```
Comment (micbar) [inline docs/ocis/adr/0006-service-discovery.md:49]: hehe, nothing good?

Comment (micbar) [reply docs/ocis/adr/0006-service-discovery.md:49]: maybe good, because a static setup is less complex?

**Review by micbar — COMMENTED** [review]
Some remarks.

File: docs/ocis/adr/0006-service-discovery.md (hunk 4 above)
Comment (refs) [reply docs/ocis/adr/0006-service-discovery.md:49]: oh I forgot. There's also the topic of firewall design, static IP addresses are joy as opposed to dynamic IP ranges.

**Review by micbar — APPROVED** [review]
LGTM

**Pull request merged**